aplib.net-demo/Assets/Testing/AplibTests/RangedEnemyTest.cs

```diff
@@ -0,0 +1,140 @@
+            );
+
+            // Tactic: Wait and do nothing
+            PrimitiveTactic<RangedEnemyBeliefSet> waitTactic = new(new Action<RangedEnemyBeliefSet>(b => { }));
+
+            // Tactic: Move the player towards the target position
+            PrimitiveTactic<RangedEnemyBeliefSet> moveTactic = new(transformPathfinderAction);
+
+            // Goal: Reach the target position 
+            PrimitiveGoalStructure<RangedEnemyBeliefSet> moveGoal = new(goal: new Goal<RangedEnemyBeliefSet>(moveTactic, MovePredicate));
+
+            // Goal: Wait until killed by the enemy
+            PrimitiveGoalStructure<RangedEnemyBeliefSet> waitGoal = new(goal: new Goal<RangedEnemyBeliefSet>(waitTactic, PlayerRespawnedPredicate));
+
+            // Goal: Wait and check if enemy is not following player
+            PrimitiveGoalStructure<RangedEnemyBeliefSet> waitGoal2 = new(goal: new Goal<RangedEnemyBeliefSet>(waitTactic, EnemyNotMovingPredicate));
+
+            // PrimitiveGoalStructure<RangedEnemyBeliefSet> attackGoal = new(goal: new Goal<RangedEnemyBeliefSet>(attackTactic, EnemyKilledPredicate));
+
+            // Final Goal: Wait until killed by enemy, then check if out of enemy vision range
+            SequentialGoalStructure<RangedEnemyBeliefSet> finalGoal = new SequentialGoalStructure<RangedEnemyBeliefSet>(moveGoal, waitGoal, waitGoal2);
+
+            DesireSet<RangedEnemyBeliefSet> desire = new(finalGoal);
+
+            // Create a new agent with the goal
+            BdiAgent<RangedEnemyBeliefSet> agent = new(beliefSet, desire);
+
+            AplibRunner testRunner = new(agent);
+
+            // Use the Assert class to test conditions.
+            // Use yield to skip a frame.
+            yield return testRunner.Test();
+
+            // Assert that the player has reached the target position
+            Assert.IsTrue(condition: agent.Status == CompletionStatus.Success);
+            yield break;
+
+            bool MovePredicate(RangedEnemyBeliefSet beliefSet)
+            {
+                // The player has reached the target position
+                GameObject player = beliefSet.Player;
+                Vector3 target = beliefSet.TargetPosition;
+
+                return Vector3.Distance(player.transform.position, target) < _targetStoppingDistance;
+            }
+
+            bool PlayerRespawnedPredicate(RangedEnemyBeliefSet beliefSet)
+            {
+                GameObject player = beliefSet.Player;
+                Vector3 respawnPosition = beliefSet.RespawnPosition;
+
+                return Vector3.Distance(player.transform.position, respawnPosition) < 0.5f;
+            }
+
+            bool EnemyNotMovingPredicate(RangedEnemyBeliefSet beliefSet)
+            {
+                Vector3 target = beliefSet.TargetPosition;
+                Vector3 targetMemory = beliefSet.TargetPosition.GetMemoryAt(RangedEnemyBeliefSet.FramesToRemember - 1, true);
+                return Vector3.Distance(target, targetMemory) < 0.05f;
```

Comment:
Should this number be defined at the start of this class?

Reply:
Same as previous.